lib/projfs.c

```diff
@@ -1451,10 +1505,66 @@ static int check_safe_rel_path(const char *path)
 	return 1;
 }
 
-int projfs_create_proj_dir(struct projfs *fs, const char *path, mode_t mode)
+#define PROJ_XATTR_READ 0x00
+#define PROJ_XATTR_WRITE 0x01
+#define PROJ_XATTR_CREATE 0x02
+
+static char *make_user_xattr_name(const char *segments, unsigned int flags)
```

Comment:
Some documentation around what/why would be good here. (the central `if` doesn't explain itself very well)